x-pack/plugins/endpoint/common/types.ts

```diff
@@ -51,6 +51,29 @@ export class EndpointAppConstants {
   static MAX_LONG_INT = '9223372036854775807'; // 2^63-1
 }
 
+export interface NodeStats {
+  totalEvents: number;
+  totalAlerts: number;
+}
+
+export interface NodePagination {
+  nextChild?: string | null;
+  nextEvent?: string | null;
+  nextAncestor?: string | null;
+  nextAlert?: string | null;
+}
+
+export interface Node {
+  id: string;
+  children: Node[];
+  events: ResolverEvent[];
+  lifecycle: ResolverEvent[];
+  ancestors?: Node[];
```

Comment:
❔ Are these in any particular order like `[closestAncestor ... farthesAncestor]`?

Reply:
Yeah, you're correct, `[closestAncestor ... farthesAncestor]`, probably need to add a comment about that